### PR TITLE
added a new data structure course in python

### PR DESCRIPTION
**Pull request by SurabhiSuresh22**
## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo

### Why is this valuable (or not)?
One of the best available free resource to learn data structures in python.

### How do we know it's really free?
Please do check the link. It is a youtube course.

### For book lists, is it a book? For course lists, is it a course? etc.
It is a course.

## Checklist:
- [ ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [ ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [ ] Include author(s) and platform where appropriate.
- [ ] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction).
- [ ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -146,6 +146,7 @@
 * [Data Structures and Algorithms Specialization](https://www.coursera.org/specializations/data-structures-algorithms) - UC San Diego, HSE University
 * [Data Structures Easy to Advanced Course - Full Tutorial from a Google Engineer](https://www.youtube.com/watch?v=RBSGKlAvoiM) - William Fiset (freeCodeCamp)
 * [Data Structures in C++ - For Beginners](https://www.udemy.com/course/data-structures-for-beginners-c-plusplus) - Pedro Mercado (Udemy)
+* [Data Structures in Python](https://www.youtube.com/playlist?list=PLzgPDYo_3xukPJdH6hVQ6Iic7KiJuoA-l) - Amulya's Academy (YouTube)
```
Comment (shrn01) [inline courses/free-courses-en.md:149]:
```suggestion
* [Data Structures \| Python](https://www.youtube.com/playlist?list=PLzgPDYo_3xukPJdH6hVQ6Iic7KiJuoA-l) - Amulya's Academy
```
Use the name provided by the resource, Youtube attribution is not required

**Review by shrn01 — CHANGES_REQUESTED** [review]

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -146,6 +146,7 @@
 * [Data Structures and Algorithms Specialization](https://www.coursera.org/specializations/data-structures-algorithms) - UC San Diego, HSE University
 * [Data Structures Easy to Advanced Course - Full Tutorial from a Google Engineer](https://www.youtube.com/watch?v=RBSGKlAvoiM) - William Fiset (freeCodeCamp)
 * [Data Structures in C++ - For Beginners](https://www.udemy.com/course/data-structures-for-beginners-c-plusplus) - Pedro Mercado (Udemy)
+* [Data Structures in Python](https://www.youtube.com/playlist?list=PLzgPDYo_3xukPJdH6hVQ6Iic7KiJuoA-l) - Amulya's Academy
```
Comment (shrn01) [inline courses/free-courses-en.md:149]:
```suggestion
* [Data Structures \| Python](https://www.youtube.com/playlist?list=PLzgPDYo_3xukPJdH6hVQ6Iic7KiJuoA-l) - Amulya's Academy
```
Do not change the resource name

**Review by shrn01 — CHANGES_REQUESTED** [review]

**Review by shrn01 — COMMENTED** [review]
Looks like the linter is failing, but is not showing proper error

**Comment by eshellman** [comment]
I think it's something with the ' character

**Comment by eshellman** [comment]
OK, it seems it was the pipe character, which apparently sorts as if it's not there. but apparently the linter can't emit a useful error message.

**Pull request merged**